tests/test_password_sign_on.py

```diff
@@ -50,100 +57,130 @@ def crazy_casing(netid: str) -> str:
     return ''.join(letters)
 
 
-# PWD 1–4, 6–8, 10
-@pytest.mark.parametrize('login_transform', [
-    lambda n: n,
-    partial(add_suffix, '@u.washington.edu'),
-    partial(add_suffix, '@washington.edu'),
-    partial(add_suffix, '@uw.edu'),
-    crazy_casing,
-])
-def test_new_session_std_sign_on(utils, browser, sp_url, sp_domain, new_tab, secrets, netid, login_transform):
-    login = login_transform(netid)
+def test_pwd_1_2_3(utils, browser, sp_url, sp_domain, new_tab, secrets, netid):
+    # 1
     password = secrets.test_accounts.password
-    sp = ServiceProviderInstance.diafine6
+    sp = ServiceProviderInstance.diafine8
     with utils.using_test_sp(sp):
         browser.get(f'{sp_url(sp)}/shibprod')
-        browser.send_inputs(login, password)
+        browser.send_inputs(netid, password)
         browser.click(Locators.submit_button)
         browser.wait_for_tag('h2', f'{sp_domain(sp)} sign-in success!')
 
-    sp = ServiceProviderInstance.diafine12
+    # 2
+    sp = ServiceProviderInstance.diafine6
     with utils.using_test_sp(sp):
         new_tab()
-        browser.get(f'{sp_url(sp)}/shibprod')
+        with browser.autocapture_off():
+            browser.get(f'{sp_url(sp)}/shibprod')
         browser.wait_for_tag('h2', f'{sp_domain(sp)} sign-in success!')
-
+    # 3 logout and close browser
+        with browser.autocapture_off():
+            browser.click_tag('a', 'Return to the main menu')
+        with browser.autocapture_off():
+            browser.wait_for_tag('h1', 'Diafine6 IdP Testing Platform')
+        browser.click_tag('a', 'Logout')
+        browser.wait_for_tag('span', 'Your UW NetID sign-in session has ended')
+        browser.close()
+
+
+def test_pwd_4(utils, browser, sp_url, secrets, netid):
+    fresh_browser = Chrome()
+    sp = ServiceProviderInstance.diafine6
+    with utils.using_test_sp(sp):
+        fresh_browser.get(f'{sp_url(sp)}/shibprod')
+        password = secrets.test_accounts.password
+        badnetid = 'non-existant-0123456789'
+        fresh_browser.send_inputs(badnetid, password)
+        fresh_browser.click(Locators.submit_button)
+        fresh_browser.wait_for_tag('p', 'Your sign-in failed.')
+        element = fresh_browser.find_element_by_id('weblogin_netid')
+        element.click()
+        element.clear()
+        badpassword = '1'
+        fresh_browser.send_inputs(netid, badpassword)
+        fresh_browser.click(Locators.submit_button)
+        fresh_browser.wait_for_tag('p', 'Your sign-in failed.')
+        fresh_browser.close()
```

Comment:
Rather than having to remember to close this each time, you can instead do:

```
with Chrome() as fresh_browser:
    with utils.using_test_sp(sp):
        # . . . 
```

Not a blocker, just an alternative if you're so inclined. Contexts are great.

Reply:
I'd probably forget that its opening and closing the browser :(